src/main/java/mods/railcraft/common/carts/EntityCartCommand.java

```diff
@@ -44,9 +44,7 @@ protected void writeEntityToNBT(NBTTagCompound compound) {
         super.writeEntityToNBT(compound);
         // Command block logic overrode this and the custom name is
         // somewhat set to the command block's name ("@")
-        if (!getCustomNameTag().isEmpty()) {
```

Comment:
Actually, I found that even if it does not have a custom name, the cart's name would be set to the command block's default name `@`.